apps/studio/components/interfaces/Integrations/Vault/Secrets/SecretRow.tsx

```diff
@@ -128,15 +130,16 @@ const SecretRow = ({ secret, onSelectEdit, onSelectRemove }: SecretRowProps) =>
                   <Trash stroke="red" size="14" />
                   <p className="text-foreground-light">Delete</p>
                 </DropdownMenuItem>
-              </TooltipTrigger>
-              {!canManageSecrets && (
-                <TooltipContent side="bottom">
-                  You need additional permissions to delete secrets
-                </TooltipContent>
-              )}
-            </Tooltip>
-          </DropdownMenuContent>
-        </DropdownMenu>
+              </DropdownMenuContent>
+            </DropdownMenu>
+          </TooltipTrigger>
+          {!canManageSecrets && (
+            <TooltipContent side="bottom">
+              You need additional permissions to manage secrets
+            </TooltipContent>
+          )}
+        </Tooltip>
```

Comment:
rather than having a repetitive message for each menu item, we can just disable the whole menu since all the items require permission to manage the secret